glyphs-reader/src/font.rs

```diff
@@ -1346,19 +1348,48 @@ impl RawFont {
         Ok(())
     }
 
-    fn v2_to_v3_weight(&mut self) -> Result<(), Error> {
+    fn v2_to_v3_master_names(&mut self) -> Result<(), Error> {
+        // in Glyphs 2, masters don't have a single 'name' attribute, but rather
+        // a concatenation of three other optional attributes weirdly called
+        // 'width', 'weight' and 'custom' (in exactly this order).
+        // The first two can only contain few predefined values, the last one is
+        // residual and free-form. They default to 'Regular' when omitted in
+        // the source. See:
+        // https://github.com/schriftgestalt/GlyphsSDK/blob/Glyphs3/GlyphsFileFormat/GlyphsFileFormatv2.md
+        // https://github.com/googlefonts/glyphsLib/blob/9d5828d/Lib/glyphsLib/classes.py#L1700-L1711
         for master in self.font_master.iter_mut() {
-            // Don't remove weightValue, we need it to understand axes
-            if master.weight_value.is_none() {
+            // Even though glyphs2 masters don't officially have a 'name' attribute,
+            // some glyphs2 sources produced by more recent versions of Glyphs
+            // sometimes have it (unclear exactly when or from which version on).
+            // We keep the 'name' attribute as is, instead of generating a one.
+            if master.name.is_some() {
                 continue;
             }
-            // Missing = default = Regular per @anthrotype
-            master.name = Some(
-                master
-                    .weight
-                    .take()
-                    .unwrap_or_else(|| String::from("Regular")),
-            );
+            let width_name = master.width.take();
+            let weight_name = master.weight.take();
+            let custom_name = master.custom.take();
+            // Remove Nones, empty strings and redundant occurrences of 'Regular'
+            let mut names: Vec<_> = vec![width_name, weight_name, custom_name]
+                .into_iter()
+                .flatten()
+                .filter(|x| !x.is_empty() && x != "Regular")
+                .collect();
+            // append "Italic" if italic angle != 0
+            if let Some(italic_angle) = master.italic_angle {
+                if italic_angle != 0.0
+                    && (names.is_empty()
+                        || (!names.contains(&"Italic".into())
+                            && !names.contains(&"Oblique".into())))
```

Comment:
```suggestion
                    && (names.is_empty()
                        || !names.iter().any(|name| name == "Italic" || name == "Oblique")
```

(avoiding allocating the two strings)